README.md

```diff
@@ -26,7 +26,41 @@ jobs:
       - run: swift test
 ```
 
+* Sample workflow using a custom Swift toolchain from a Github repository
+
+```yaml
+on: [pull_request]
+
+jobs:
+  windows:
+    runs-on: windows-latest
+    steps:
+      - uses: compnerd/gha-setup-swift@main
+        with:
+          branch: main
+          tag: tags/20230530.2
```

Comment:
Can make this `refs/tags/20230503.2` and accept that style?

Reply:
I am not sure, this is what https://github.com/dsaltares/fetch-gh-release-asset#version takes.

Reply:
Oh, that is a parameter passed to the task.  I wonder if we can simply interposition the `tags/` prefix then rather than requiring it?

Reply:
Not really, because it takes either tags/TAG or the release ID (which I think it is the same as TAG in most cases, but I'd think it can be different)

Reply:
My previous comment was confusing and factually incorrect.

The github API for fetching releases uses one of two methods: `tag_name` or `release_id`. The former is what it says, the actual git tag of the release ("v5.2.1" for example), and is reflected in the git tags as `refs/tags/<tagname>`, while the latter is a unique github only integer that identifies the release for Github and is not reflected in git.

These are the Github API calls to fetch a release:
GET /repos/{owner}/{repo}/releases/{release_id}
GET /repos/{owner}/{repo}/releases/tags/{tag}

```
$ git ls-remote --tags origin "*v5.2.1"
51a1a9421c23a3dba8185734c3fca415b7ed1059	refs/tags/v5.2.1
```

The Github action I use to fetch releases reflects this in a bit of a convoluted way:
version: The release version to fetch from in the form tags/<tag_name> or <release_id>. Defaults to latest.

So, one passes `latest`, `tags/<tag_name>` or `<release_id>` to it.

Here's my suggestion to make this more correct and clear, please let me know if that's fine for you:

```
// official builds from swift.org. No changes so it doesn't break current users.
      - uses: compnerd/gha-setup-swift@main
        with:
          branch: swift-5.5-release
          tag: 5.5-RELEASE

// custom builds: (branch is not needed at all, and I believe we don't need to support release_id, the github-specific integer
      - uses: compnerd/gha-setup-swift@main
        with:
          github-repo: mycompany/swift-toolchain-build
          release-tag-name: 20230530.2
          release-asset-name: installer-amd64.exe
```

Reply:
I think that looks wonderful!  Thank you.